Controllers/OperationController.cs

```diff
@@ -99,6 +99,9 @@ public class OperationController : ApiController
         [Authorize]
         public async Task<HttpResponseMessage> Invoke(OperationInfo info)
         {
+            // escapeuristring does not escape '#' as it is a valid special character
+            // will not be legit in our case so escaping explicitly.
+            info.Url = Uri.EscapeUriString(info.Url).Replace("#", "%23");
```

Comment:
ok final change will look like this 

// escaping "#" as it may appear in some resource names
            info.Url = info.Url.Replace("#", Uri.EscapeDataString("#"));

makes sense?